multiversx_sdk/converters/transactions_converter_test.py

```diff
@@ -47,3 +65,147 @@ def test_transaction_converter():
+    logs.events = [event]
+
+    contract_result_item = ContractResultItemOnNetwork()
+    contract_result_item.nonce = 8
+    contract_result_item.data = "@6f6b@2a"
+    contract_result_item.logs = logs
+
+    contract_result = ContractResultOnNetwork([contract_result_item])
+    tx_on_network.contract_results = contract_result
+
+    actual_tx_outcome = converter.transaction_on_network_to_outcome(tx_on_network)
+
+    expected_tx_outcome = TransactionOutcome(
+        transaction_results=[SmartContractResult(
+            sender="",
+            receiver="",
+            data=b"@6f6b@2a",
+            logs=TransactionLogs(
+                address="erd1k2s324ww2g0yj38qn2ch2jwctdy8mnfxep94q9arncc6xecg3xaq6mjse8",
+                events=[TransactionEvent(
+                    address="",
+                    identifier="writeLog",
+                    topics=[b"@too much gas provided for processing: gas provided = 596384500, gas used = 733010"],
+                    data_items=[base64.b64decode("QDZmNmI=")]
+                )]
+            )
+        )],
+        transaction_logs=TransactionLogs(
+            address="erd1k2s324ww2g0yj38qn2ch2jwctdy8mnfxep94q9arncc6xecg3xaq6mjse8",
+            events=[
+                TransactionEvent(
+                    address="",
+                    identifier="foobar",
+                    data_items=[b"foo"]
+                )]
+        )
+    )
+
+    assert actual_tx_outcome.logs.address == expected_tx_outcome.logs.address
+    assert actual_tx_outcome.logs.events[0].identifier == expected_tx_outcome.logs.events[0].identifier
+    assert actual_tx_outcome.logs.events[0].data_items == expected_tx_outcome.logs.events[0].data_items
+    assert actual_tx_outcome.logs.events[0].address == expected_tx_outcome.logs.events[0].address
+    assert actual_tx_outcome.logs.events[0].topics == expected_tx_outcome.logs.events[0].topics
+
+    assert actual_tx_outcome.transaction_results[0].sender == expected_tx_outcome.transaction_results[0].sender
+    assert actual_tx_outcome.transaction_results[0].receiver == expected_tx_outcome.transaction_results[0].receiver
+    assert actual_tx_outcome.transaction_results[0].data == expected_tx_outcome.transaction_results[0].data
+    assert actual_tx_outcome.transaction_results[0].logs.address == expected_tx_outcome.transaction_results[0].logs.address
+    assert actual_tx_outcome.transaction_results[0].logs.events[0].address == expected_tx_outcome.transaction_results[0].logs.events[0].address
+    assert actual_tx_outcome.transaction_results[0].logs.events[0].identifier == expected_tx_outcome.transaction_results[0].logs.events[0].identifier
+    assert actual_tx_outcome.transaction_results[0].logs.events[0].data_items == expected_tx_outcome.transaction_results[0].logs.events[0].data_items
+    assert actual_tx_outcome.transaction_results[0].logs.events[0].topics == expected_tx_outcome.transaction_results[0].logs.events[0].topics
```

Comment:
No way to do a shorter assert? Deep equality etc.?

Reply:
No "native" way as far as I know.  Will think about doing a custom `deep_equal()` method to use in some tests.